game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/firststrike/DefensiveFirstStrikeTest.java

```diff
@@ -128,4 +114,27 @@ void getStep(final List<BattleStateVariation> parameters, final BattleStep.Order
                 BattleStateVariation.HAS_DEFENDING_SUBS_SNEAK_ATTACK),
             Order.FIRST_STRIKE_DEFENSIVE));
   }
+
+  @Test
+  void firingFilterOnlyFindsFirstStrikeUnits() {
+
+    final GameData gameData = givenGameData().build();
+
+    final List<Unit> units =
+        List.of(
+                givenAnyUnit(),
+                givenUnitFirstStrike(),
+                givenAnyUnit(),
+                givenUnitFirstStrike(),
+                givenAnyUnit())
+            .stream()
+            .filter(DefensiveFirstStrike.FIRING_UNIT_PREDICATE.apply(gameData))
+            .collect(Collectors.toList());
+
+    assertThat(units, hasSize(2));
+    assertThat(
+        "There should be only first strike units",
+        units.stream().allMatch(Matches.unitIsFirstStrike()),
+        Matchers.is(true));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.is'](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6277681)